src/core/cancellation_middleware.py

```
@@ -24,16 +24,20 @@ async def on_call_tool(self, context: MiddlewareContext, call_next):
+            
             result = await call_next(context)
             
             duration_ms = (time.perf_counter() - start_time) * 1000
-            logger.debug(f"Tool call completed in {duration_ms:.2f}ms")
+            logger.info(f"✅ CANCELLATION MIDDLEWARE: Tool call completed successfully in {duration_ms:.2f}ms")
             return result
             
         except asyncio.CancelledError:
             duration_ms = (time.perf_counter() - start_time) * 1000
-            logger.warning(f"Tool call cancelled after {duration_ms:.2f}ms")
+            logger.error(f"❌ CANCELLATION MIDDLEWARE: Tool call cancelled after {duration_ms:.2f}ms")
```

Ruff (0.12.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)